packages/salesforcedx-vscode-core/src/commands/forceApexExecute.ts

```diff
@@ -109,10 +109,14 @@ export class AnonApexGatherer
       }
 
       const document = editor.document;
-      if (!editor.selection.isEmpty) {
+      if (!editor.selection.isEmpty || !fs.existsSync(document.uri.fsPath)) {
```

Comment:
can we check for `document.isUntitled` here instead?

Reply:
Oh so cool, I haven't seen that api before